FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -0,0 +1,1142 @@
+        {
+            /* Ensure that the buffer pointer is accurate after being copied from the queue. */
+            xResponseItem.xResponse.pBuffer = xResponseItem.ucResponseBuffer;
+
+            /* Ensure that we received a successful response from the server. */
+            if( xResponseItem.xResponse.statusCode != httpexampleHTTP_STATUS_CODE_PARTIAL_CONTENT )
+            {
+                LogError( ( "Received response with unexpected status code: %d.", xResponseItem.xResponse.statusCode ) );
+                xStatus = pdFAIL;
+            }
+            else
+            {
+                xHTTPStatus = HTTPClient_ReadHeader( &xResponseItem.xResponse,
+                                                     ( char * ) httpexampleHTTP_CONTENT_RANGE_HEADER_FIELD,
+                                                     ( size_t ) httpexampleHTTP_CONTENT_RANGE_HEADER_FIELD_LENGTH,
+                                                     ( const char ** ) &pcContentRangeValStr,
+                                                     &xContentRangeValStrLength );
+
+                if( xHTTPStatus != HTTPSuccess )
+                {
+                    LogError( ( "Failed to read Content-Range header from HTTP response: Error=%s.",
+                                HTTPClient_strerror( xHTTPStatus ) ) );
+                    xStatus = pdFAIL;
+                }
+            }
+
+            break;
+        }
+    }
+
+    /* Parse the Content-Range header value to get the file size. */
+    if( xStatus == pdPASS )
+    {
+        pcFileSizeStr = strstr( pcContentRangeValStr, "/" );
+
+        if( pcFileSizeStr == NULL )
+        {
+            LogError( ( "'/' not present in Content-Range header value: %s.",
+                        pcContentRangeValStr ) );
+            xStatus = pdFAIL;
+        }
+    }
+
+    if( xStatus == pdPASS )
+    {
+        pcFileSizeStr += sizeof( char );
+        xFileSize = ( size_t ) strtoul( pcFileSizeStr, NULL, 10 );
+
+        if( ( xFileSize == 0 ) || ( xFileSize == UINT32_MAX ) )
+        {
+            LogError( ( "Error using strtoul to get the file size from %s: xFileSize=%d.",
+                        pcFileSizeStr, ( int32_t ) xFileSize ) );
+            xStatus = pdFAIL;
+        }
+    }
+
+    if( xStatus == pdPASS )
+    {
+        LogInfo( ( "The file is %d bytes long.", ( int32_t ) xFileSize ) );
+    }
+
+    return xStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvResponseTask( void * pvArgs )
+{
+    uint32_t ulWaitCounter = 0U;
+    uint32_t ulNotification = 0U;
+    size_t xNumResponses = 0;
+
+    ( void ) pvArgs;
+
+    if( prvReadFileSize() != pdPASS )
+    {
+        LogError( ( "Response task: Error obtaining file size from the server response. Exiting task." ) );
+
+        /* Notify the other tasks of failure. */
+        xTaskNotify( xRequestTask, httpexampleHTTP_FAILURE, eSetBits );
```

Comment:
Does Request task monitor this flag?

Reply:
The request task checks for this notification while the file size value is still 0 (in case of an error with the server response), essentially waiting for the response task to determine the file size before proceeding.